taxcalc/behavior.py

```diff
@@ -407,6 +476,151 @@ def _validate_parameter_values(self):
+                        # pylint: disable=consider-using-enumerate
+                        for idx in range(0, len(pvalue)):
+                            if scalar:
+                                pname = name
+                            else:
+                                pname = '{}_{}'.format(name, idx)
+                            pval = pvalue[idx]
+                            # pylint: disable=unidiomatic-typecheck
+                            pval_is_bool = type(pval) == bool
+                            pval_is_int = type(pval) == int
+                            pval_is_float = type(pval) == float
+                            if bool_param_type:
+                                if not pval_is_bool:
+                                    msg = '{} {} value {} is not boolean'
+                                    self.parameter_errors += (
+                                        'ERROR: ' +
+                                        msg.format(year, pname, pval) +
+                                        '\n'
+                                    )
+                            elif int_param_type:
+                                if not pval_is_int:
+                                    msg = '{} {} value {} is not integer'
+                                    self.parameter_errors += (
+                                        'ERROR: ' +
+                                        msg.format(year, pname, pval) +
+                                        '\n'
+                                    )
+                            else:  # param is float type
+                                if not (pval_is_int or pval_is_float):
+                                    msg = '{} {} value {} is not a number'
+                                    self.parameter_errors += (
+                                        'ERROR: ' +
+                                        msg.format(year, pname, pval) +
+                                        '\n'
+                                    )
+        del param_names
+
+    def _validate_parameter_values(self, parameters_set):
+        """
+        Check values of parameters in specified parameter_set using
+        range information from the current_law_policy.json file.
+        """
+        # pylint: disable=too-many-locals
+        # pylint: disable=too-many-branches
+        # pylint: disable=too-many-nested-blocks
+        rounding_error = 100.0
+        # above handles non-rounding of inflation-indexed parameter values
+        clp = self.baseline_version()
+        parameters = sorted(parameters_set)
+        syr = Behavior.JSON_START_YEAR
+        for pname in parameters:
+            if pname.endswith('_cpi'):
+                continue  # *_cpi parameter values validated elsewhere
+            pvalue = getattr(self, pname)
+            for vop, vval in self._vals[pname]['range'].items():
+                if isinstance(vval, six.string_types):
```

Comment:
@martinholmer this condition is true if you have a range field in `behavior.json` that looks something like `"range": {"min": "param name", "max": ...}`, right?

If this is correct, do you think we should remove this case? Then if we need it some time in the future, we could just add it back.

Reply:
OK, remove the code.